homeassistant/components/livisi/switch.py

```diff
@@ -92,149 +85,99 @@ def __init__(
         self._attr_unique_id = unique_id
         self._manufacturer = manufacturer
         self._product = product
         self._serial_number = serial_number
-        self._device_type = device_type
-        self._name = name
+        self._attr_model = device_type
+        self._attr_name = name
         self._state = None
         self._capability_id = capability_id
-        self._is_on = is_on
+        self._attr_is_on = is_on
         self._room = room
         self._version = version
-        self.aio_livisi = AioLivisi.get_instance()
+        self.aio_livisi = shc.aiolivisi
         if is_on is None:
-            self._is_available = False
+            self._attr_is_available = False
         else:
-            self._is_available = True
-
-    @property
-    def manufacturer(self) -> str:
-        """Return the manufacturer."""
-        return self._manufacturer
-
-    @manufacturer.setter
-    def manufacturer(self, new_value: str):
-        self._manufacturer = new_value
-
-    @property
-    def product(self) -> str:
-        """Return the product type."""
-        return self._product
-
-    @product.setter
-    def product(self, new_value: str):
-        self._product = new_value
-
-    @property
-    def serial_number(self) -> str:
-        """Return the serial number."""
-        return self._serial_number
-
-    @serial_number.setter
-    def serial_number(self, new_value: str):
-        self._serial_number = new_value
-
-    @property
-    def device_type(self) -> str:
-        """Return the device type."""
-        return self._device_type
-
-    @device_type.setter
-    def device_type(self, new_value: str):
-        self._device_type = new_value
-
-    @property
-    def name(self):
-        """Return the name of the device."""
-        return self._name
-
-    @name.setter
-    def name(self, new_value: str):
-        self._name = new_value
-
-    @property
-    def version(self):
-        """Return the version number."""
-        return self._version
-
-    @version.setter
-    def version(self, new_value: str):
-        self._version = new_value
-
-    @property
-    def capability_id(self):
-        """Return the capability id of the device."""
-        return self._capability_id
+            self._attr_is_available = True
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(self._attr_unique_id))},
+            manufacturer=self._manufacturer,
+            model=self._attr_model,
+            name=self._attr_name,
+            suggested_area=self._room,
+            sw_version=self._version,
+            via_device=(DOMAIN, str(self._attr_unique_id)),
+        )
 
-    @capability_id.setter
-    def capability_id(self, new_value: str):
-        self._capability_id = new_value
+        config_entry.async_on_unload(
+            async_dispatcher_connect(hass, LIVISI_STATE_CHANGE, self.update_states)
+        )
+        config_entry.async_on_unload(
+            async_dispatcher_connect(
```

Comment:
Please register update callbacks in `async_added_to_hass` entity coroutine function.